### PR TITLE
SAIC-170 Start ssh-agent once for each private key

### PR DESCRIPTION
**Pull request by roman-verchikov**
ssh-agent now starts for the first ssh command issued and never gets
killed. This is done in order to improve end-user (deployment engineer)
experience in case he/she has password-protected private key. With this
patch enabled password is only required to be entered once for the whole
migration process.

### CODE REVIEW

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

File: cloudferrylib/utils/utils.py [hunk 1]
```diff
@@ -296,27 +296,45 @@ def inner(*args, **kwargs):
...
         self.ssh_auth_sock = info_agent[1]
         os.environ["SSH_AGENT_PID"] = self.pid
         os.environ["SSH_AUTH_SOCK"] = self.ssh_auth_sock
 
     def __exit__(self, type, value, traceback):
-        local("kill -9 %s"%(self.pid))
-        del os.environ["SSH_AGENT_PID"]
-        del os.environ["SSH_AUTH_SOCK"]
+        # never kill previously started ssh-agent, so that user only has to
+        # enter private key password once
```
Comment (miarmak) [inline cloudferrylib/utils/utils.py]: Does it mean, that in the end of migration we will have ssh-agent launched?

**Comment by miarmak** [comment]
Looks good and works for me. Have little question inline.

File: cloudferrylib/utils/utils.py (hunk 1 above)
Comment (roman-verchikov) [inline cloudferrylib/utils/utils.py]: Yes, exactly that. And I don't see any problem with that. Actually I think ssh-agent should be started prior to running cloudferry.

**Comment by Wasapl** [comment]
Merged build finished. Test PASSed.

File: cloudferrylib/utils/utils.py (hunk 1 above)
Comment (miarmak) [inline cloudferrylib/utils/utils.py]: Ok

**Comment by miarmak** [comment]
+1

**Pull request merged**